docs/rules/replace-j-unit4-assert-with-jupiter.md

```diff
@@ -0,0 +1,177 @@
+* [`fail`](https://javadoc.io/doc/junit/junit/latest/org/junit/Assert.html#fail())
+
+and replaces them with the invocations of the corresponding - also overloaded - alternatives declared in the JUnit 5 class Assertions:
+
+* [`assertArrayEquals`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertArrayEquals(java.lang.Object%5B%5D,java.lang.Object%5B%5D))
+* [`assertEquals`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertEquals(java.lang.Object,java.lang.Object))
+* [`assertFalse`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertFalse(boolean))
+* [`assertNotEquals`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertNotEquals(java.lang.Object,java.lang.Object))
+* [`assertNotNull`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertNotNull(java.lang.Object))
+* [`assertNotSame`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertNotSame(java.lang.Object,java.lang.Object))
+* [`assertNull`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertNull(java.lang.Object))
+* [`assertSame`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertSame(java.lang.Object,java.lang.Object))
+* [`assertTrue`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertTrue(boolean))
+* [`fail`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#fail())
+
+By replacing each of these JUnit 4 Assert method invocations by the corresponding Jupiter alternatives, this rule promotes a stepwise transition to JUnit Jupiter.
+
+::: warning Requirements
+This rule requires the following library to be present:
+* junit:junit:4.13
+* org.junit.jupiter:junit-jupiter-api:5.4.0
+:::
+
+## Benefits
+
+Migrates JUnit 4 tests to JUnit 5.
+
+## Tags
+
+::: tip Tags
+<TagLinks />
+:::
+
+## Code Changes
+
+
+### Replacing Imports
+
+__Pre__
+```java
+package sim1892.junit.jupiter.githubdoc.replacingimport;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class TestExample {
+	@Test
+	void test() {
+		assertEquals(0L, 0L);
+	}
+}
+```
+
+__Post__
+```java
+package sim1892.junit.jupiter.githubdoc.replacingimport;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestExample {
+	@Test
+	void test() {
+		assertEquals(0L, 0L);
+	}
+}
+```
+
+### Replacing Invocations with a Message
+
+__Pre__
+```java
+package sim1892.junit.jupiter.githubdoc.usingmessage;
```

Comment:
Please be careful to not expose internal information like ticket numbers, or package names.